updated dependencies needed for sphinx docs

orbeckst opened the pull request:
- fix #260
- updated required dependencies for sphinx docs (especially the bibtex packages)
- moved some packages from pip to conda

orbeckst committed:
updated dependencies needed for sphinx docs

- fix #260
- updated required dependencies for sphinx docs (especially the bibtex packages)
- moved some packages from pip to conda

IAlibay commented on doc/source/contributing_code.rst:
```diff
@@ -217,11 +217,13 @@ First we need to install dependencies. You'll need a mix of conda and pip instal
             griddataformats gsd hypothesis "joblib>=0.12" \
             matplotlib mmtf-python mock netcdf4 networkx \
             "numpy>=1.18.0" psutil pytest scikit-learn scipy \
```
we probably should update the numpy minimum pin too

orbeckst replied:
yes, I just wanted to keep the commit focused

orbeckst replied:
will update to whatever I find in pyproject.toml

IAlibay replied:
If you want we can just merge this and I can just do a cleanup afterwards. There's probably a bunc of stuff in the page that needs changing.

orbeckst replied:
Just saw your comment _after_ I did a bunch of things. If you think it's useful, merge and then we can update later.

IAlibay replied:
I think you've done more or less all the changes I think needed doing. I don't think we need two PRs. (it was mostly a "if you're kinda busy we can push it to the next PR")

orbeckst committed:
updated dependencies from pyproject.toml

orbeckst commented on doc/source/contributing_code.rst:
```diff
@@ -212,18 +214,37 @@ First we need to install dependencies. You'll need a mix of conda and pip instal
+	  'pyedr>=0.7.0' \
+	  'pytng>=0.2.3' \
+	  'gsd>3.0.0' \
+	  'rdkit>=2020.03.1' \
```
not yet in pyproject.toml https://github.com/MDAnalysis/mdanalysis/issues/4185

orbeckst commented on doc/source/contributing_code.rst:
```diff
@@ -264,16 +285,38 @@ Install the dependencies:
-          psutil pytest scikit-learn scipy "seaborn>=0.7.0" \
-          sphinx sphinx_rtd_theme "tidynamics>=1.0.0" \
-          "tqdm>=4.43.0"
+	  'Cython>=0.28' \
```
It seems a bit superfluous to explicitly pip-install all of this, given that I can just do
```bash
pip install -E package/
```
and then be assured to get the correct versions from pyproject.toml. Should we just recommend users do that? 

This would also reduce the amount of text we have to keep updated.

@RMeli @IAlibay @lilyminium ?

IAlibay replied:
Yeah although we probably should do a pip install with all the extras?

orbeckst replied:
Yes, just forgot how to do it... was it something like
```
pip install .[extra-formats,analysis,doc]
```

IAlibay replied:
`pip install ".[extra-formats,analysis,doc]"` does the trick on my end

orbeckst replied:
I don't have time to rewrite the section and test that it all works. Just not sure if there's a good reason to split dependency installation and package building.

Sorry, going to punt here.

IAlibay replied:
Pretty sure it's fine to not split but we can deal with it later

orbeckst commented on doc/source/contributing_code.rst:
```diff
@@ -212,18 +214,37 @@ First we need to install dependencies. You'll need a mix of conda and pip instal
-            "numpy>=1.18.0" psutil pytest scikit-learn scipy \
-            "seaborn>=0.7.0" sphinx "tidynamics>=1.0.0" \
-            "tqdm>=4.43.0"
+        conda install -c conda-forge \
```
I think it would be more convenient to use an environment.yaml file from the repo. Then we only need to keep _that_ file up to date.

IAlibay replied:
I think there was a discussion about it on core at some point. The main thing was that we had a ton of different files lying around that didn't make sense (for example the RTD yaml file is pinned to different version numbers because of reasons I've not had a time to dig into yet).

What we could probably do is have a standard user facing one and have CI validate it (by picking up the setup-deps action in some way or another).

orbeckst replied:
Along the lines of "check if developer.yaml is compatible with the env that we are building via setup-deps"? Sounds like a good CI check.

Perhaps just build the env with setup-deps, then try to run mamba with the yaml file and check that _nothing_ actually gets updated by diffing the env list before and after.

IAlibay replied:
yeah something like that should work

IAlibay commented on doc/source/contributing_code.rst:
```diff
@@ -85,8 +85,10 @@ touching existing environments that may have stable versions of MDAnalysis. :
     .. code-block:: bash
 
-        conda create --name mdanalysis-dev
+        conda create --name mdanalysis-dev python==3.10
```
why pin to 3.10?

orbeckst replied:
Because the only other place that a Python version is mentioned is somewhere far down (">= 3.9") and when you get to that part, it's too late to change your environment.

I wanted to add a sentence to the docs along "Look here to find out the minimum Python version currently supported by MDA and then use a Python version greater or equal to it" but I actually didn't really know what a good place to link to was. Writing "look in pyproject.yaml" seemed very off-putting, especially as we send beginners here.

If you have an idea then let me know.

I admittedly decided that I didn't have enough time to rewrite everything here (I ought to be finishing a paper or sleep).

IAlibay replied:
Could we just do >3.9 here (or 3.10 of we're worried about longevity)?

orbeckst replied:
sure, I didn't think about using a `>`.

orbeckst replied:
Will use "python>=3.9" for now, that's consistent with everything else.

orbeckst committed:
added parmed to dependencies

- see https://github.com/MDAnalysis/mdanalysis/issues/4185#issuecomment-1617569319
- untabified file

orbeckst committed:
install python>=3.9 for develop env
